base/compiler/ssair/EscapeAnalysis/EscapeAnalysis.jl

```diff
@@ -0,0 +1,1471 @@
+                    end
+                elseif head === :foreigncall
+                    escape_foreigncall!(astate, pc, stmt.args)
+                elseif head === :throw_undef_if_not # XXX when is this expression inserted ?
+                    add_escape_change!(astate, stmt.args[1], ThrownEscape(pc))
+                elseif is_meta_expr_head(head)
+                    # meta expressions doesn't account for any usages
+                    continue
+                elseif head === :enter || head === :leave || head === :the_exception || head === :pop_exception
+                    # ignore these expressions since escapes via exceptions are handled by `escape_exception!`
+                    # `escape_exception!` conservatively propagates `AllEscape` anyway,
+                    # and so escape information imposed on `:the_exception` isn't computed
+                    continue
+                elseif head === :static_parameter ||  # this exists statically, not interested in its escape
+                       head === :copyast ||           # XXX can this account for some escapes?
+                       head === :undefcheck ||        # XXX can this account for some escapes?
+                       head === :isdefined ||         # just returns `Bool`, nothing accounts for any escapes
+                       head === :gc_preserve_begin || # `GC.@preserve` expressions themselves won't be used anywhere
+                       head === :gc_preserve_end      # `GC.@preserve` expressions themselves won't be used anywhere
+                    continue
+                else
+                    for x in stmt.args
+                        add_escape_change!(astate, x, ⊤)
+                    end
+                end
+            elseif isa(stmt, ReturnNode)
+                if isdefined(stmt, :val)
+                    add_escape_change!(astate, stmt.val, ReturnEscape(pc))
+                end
+            elseif isa(stmt, PhiNode)
+                escape_edges!(astate, pc, stmt.values)
+            elseif isa(stmt, PiNode)
+                escape_val_ifdefined!(astate, pc, stmt)
+            elseif isa(stmt, PhiCNode)
+                escape_edges!(astate, pc, stmt.values)
+            elseif isa(stmt, UpsilonNode)
+                escape_val_ifdefined!(astate, pc, stmt)
+            elseif isa(stmt, GlobalRef) # global load
+                add_escape_change!(astate, SSAValue(pc), ⊤)
+            elseif isa(stmt, SSAValue)
+                escape_val!(astate, pc, stmt)
+            elseif isa(stmt, Argument)
+                escape_val!(astate, pc, stmt)
+            else # otherwise `stmt` can be GotoNode, GotoIfNot, and inlined values etc.
+                continue
+            end
+
+            isempty(changes) && continue
+
+            anyupdate |= propagate_changes!(estate, changes)
+
+            empty!(changes)
+        end
+
+        tryregions !== nothing && escape_exception!(astate, tryregions)
```

Comment:
That would work. But I don't think `compute_tryregions` is expensive since it does a simpler linear scan over statements, and also, we will often have some `ThrownEscape` anyway since the nothrow check is inherently very conservative.